### ISSUE TITLE
item.metadata assignment fails for strings longer than 32,767 characters

### ISSUE THREAD

**Issue by jacobdadams** [Esri/arcgis-python-api#611]
**Describe the bug**
The `item.metadata = ...` assignment fails for strings longer than 32,767 characters due to `os.path.isfile()` and NTFS limitations. 

In `\arcgis\gis\__init__.py`'s `metadata()` method on line 7484, it checks if the value passed is a file using `os.path.isfile()`. This check will fail with `ValueError: stat: path too long for Windows` if your value is >32,767 characters long, probably due to the limitations of NTFS (https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#maximum-path-length-limitation). This exception gets propagated up to the user, and to a user who passed a string instead of a path, this error is confusing.

**To Reproduce**
```python
#: item is an AGOL item object.
item.metadata = 'a'*32768
```
error:
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-22-78d8b8a66c57> in <module>
----> 1 item.metadata = ('a'*32768)

~\AppData\Local\Programs\ArcGIS\Pro\bin\Python\envs\validate\lib\site-packages\arcgis\gis\__init__.py in metadata(self, value)
   7482         if os.path.isfile(xml_file) == True:
   7483             os.remove(xml_file)
-> 7484         if os.path.isfile(value) == True and \
   7485            str(value).lower().endswith('.xml'):
   7486             if os.path.basename(value).lower() != 'metadata.xml':

~\AppData\Local\Programs\ArcGIS\Pro\bin\Python\envs\validate\lib\genericpath.py in isfile(path)
     28     """Test whether a path is a regular file"""
     29     try:
---> 30         st = os.stat(path)
     31     except OSError:
     32         return False

ValueError: stat: path too long for Windows
```
**Screenshots**
n/a

**Expected behavior**
High-level, I would expect the string to be assigned to the metadata property. Specific to the root cause, I would expect the ArcGIS API to provide a useful exception itself instead of having to hunt down an error propagated from a lower-level library. 

**Platform (please complete the following information):**
 - OS: Windows 10
 - Browser: n/a
 - Python API Version : `1.7.0`

**Additional context**
I discovered the error while trying to use the new `arcpy.metadata` library to copy metadata up to AGOL like thus: `agol_item.metadata = arcpy_metadata_object.xml` If there's any base64-encoded content in the metadata, it can exceed 32,767 characters pretty quickly.

Also, I first reported this through tech support as case #02499263 before I discovered I could submit it as a github issue.

**Comment by achapkowski** [comment]
@jacobdadams can you post your XML file and a feature class (zip it) that I can publish to associate with it?  I need to see what is causing the issue ie is it a Python API issue or REST issue.

If you don't/can't publish it here let me know and I can provide you with my email.

**Comment by jacobdadams** [comment]
Here's the metadata [SGID.CADASTRE.Parcels_Davis.txt](https://github.com/Esri/arcgis-python-api/files/4252403/SGID.CADASTRE.Parcels_Davis.txt) (exported via arcpy.metadata's `saveAsXML()`, .xml changed to .txt for github) and [DavisParcels.gdb.zip](https://github.com/Esri/arcgis-python-api/files/4252392/DavisParcels.gdb.zip) as requested. We are aware that this metadata has a lot of old cruft in it.

**Comment by nanaeaubry** [comment]
This issue was fixed in 2.0.1 release
